app/workers/certificates.php

```diff
@@ -26,169 +28,224 @@ public function init(): void
 
     public function run(): void
     {
-        $dbForConsole = $this->getConsoleDB();
-
-        /**
-         * 1. Get new domain document - DONE
-         *  1.1. Validate domain is valid, public suffix is known and CNAME records are verified - DONE
-         * 2. Check if a certificate already exists - DONE
-         * 3. Check if certificate is about to expire, if not - skip it
-         *  3.1. Create / renew certificate
-         *  3.2. Update loadblancer
-         *  3.3. Update database (domains, change date, expiry)
-         *  3.4. Set retry on failure
-         *  3.5. Schedule to renew certificate in 60 days
-         */
-
         Authorization::disable();
 
-        // Args
-        $document = $this->args['document'];
-        $domain = $this->args['domain'];
-
-        // Validation Args
-        $validateTarget = $this->args['validateTarget'] ?? true;
-        $validateCNAME = $this->args['validateCNAME'] ?? true;
-
-        // Options
-        $domain = new Domain((!empty($domain)) ? $domain : '');
-        $expiry = 60 * 60 * 24 * 30 * 2; // 60 days
-        $safety = 60 * 60; // 1 hour
-        $renew  = (\time() + $expiry);
-
-        if (empty($domain->get())) {
-            throw new Exception('Missing domain');
-        }
-
-        if (!$domain->isKnown() || $domain->isTest()) {
-            throw new Exception('Unknown public suffix for domain');
-        }
-
-        if ($validateTarget) {
-            $target = new Domain(App::getEnv('_APP_DOMAIN_TARGET', ''));
+        $dbForConsole = $this->getConsoleDB();
 
-            if(!$target->isKnown() || $target->isTest()) {
-                throw new Exception('Unreachable CNAME target ('.$target->get().'), please use a domain with a public suffix.');
+        $certificate = new Document();
+
+        try {
+            /**
+             * TODO: Update
+             * 1. Get new domain document - DONE
+             *  1.1. Validate domain is valid, public suffix is known and CNAME records are verified - DONE
+             * 2. Check if a certificate already exists - DONE
+             * 3. Check if certificate is about to expire, if not - skip it
+             *  3.1. Create / renew certificate
+             *  3.2. Update loadblancer
+             *  3.3. Update database (domains, change date, expiry)
+             *  3.4. Set retry on failure
+             *  3.5. Schedule to renew certificate in 60 days
+             */
+    
+    
+            // Get attributes
+            $domain = $this->args['domain']; // String of domain (hostname)
+            $domain = new Domain((!empty($domain)) ? $domain : '');
+
+            $certificate->setAttribute('domain', $domain->get());
+    
+            $skipRenewCheck = $this->args['skipRenewCheck'] ?? false; // If true, we won't double-check expiry from cert file
+    
+            $mainDomain = null; // ENV or first ever visited domain
```

Comment:
Why do we have this main domain logic here? Can we simplify the worker and treat all domains equally?

Reply:
We cant be looking for CNAME on the main domain because there you have A record.